### DISCUSSION TITLE
Capa Rust version

### DISCUSSION THREAD

**Discussion by marirs** [mandiant/capa#881] [Show and tell]
Hi,
Thanks for this wonderful project of yours. It is awesome.

I would like to mention, that I wrote portion of this into Rust. At the moment it can only extract the capabilities. No IDA plugins etc.. 

Let me know what you think: [https://github.com/marirs/capa-rs](https://github.com/marirs/capa-rs)

Thanks
Sriram

**Comment by williballenthin** [comment]
Wow @marirs nice job! Its really neat to see the work you put into this. What are some things you learned along the way? And any suggestions for capa?

In my opinion, we've found that Python was a good choice as we experimented and figured out if capa was a good idea. Now that we know the idea works, and we start to run capa against many files, we notice some limitations of Python (especially around performance). Do you have a sense for how `capa-rs` performance w.r.t CPU and memory usage?

**Comment by marirs** [comment]
Thanks @williballenthin -  the rust version would have not been there if this python wasnt there :) Loved this work. one of a kind it is. I have a static analyzer microservice - `https://analyze.rs` (a homepage to come soon) which i use for my sandbox `anlyz.io`.
the analyze.rs is completely Rust based and so the inspiration to write this in Rust came. Ofcourse a lot learnt along the way. I love the Rust performance.

`capa-rs` with mimikatz file took 11-12 seconds to analyze and 0% CPU with 100MB memory. I'm still working on more cleanup both in `smda-rs` as well as `capa-rs`. trying to see if i can achieve more performance.

**Comment by mr-tz** [comment]
Fantastic work! Thanks for sharing with us.
The performance appears very intriguing. I'd be super curious to see a larger performance comparison on speed, memory, but also extracted features, rule hits, etc.
